streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -812,7 +877,79 @@ public void closeClean() {
         assertThat(thrown.getCause().getMessage(), equalTo(null));
     }
 
+    @Test
+    public void shouldThrowTaskMigratedExceptionOnCommitFailed() {
+        final StateMachineTask task01 = new StateMachineTask(taskId01, taskId01Partitions, true);
+        task01.setCommitNeeded();
+        taskManager.tasks().put(taskId01, task01);
+
+        consumer.commitSync(new HashMap<>());
+        expectLastCall().andThrow(new CommitFailedException());
+        replay(consumer);
+
+        final TaskMigratedException thrown = assertThrows(
+            TaskMigratedException.class,
+            () -> taskManager.commitAll()
+        );
+
+        assertThat(thrown.getMessage(), equalTo("Consumer committing offsets failed, indicating the corresponding thread is no longer part of the group; it means all tasks belonging to this thread should be migrated."));
```

Comment:
Same here, for verifying the thrown cause

Reply:
This is a meta comment: since we moved the commit logic out of the tasks into task-manager already, we should add the check that:

1) inside the task manager, if the commit failed with fatal errors, the corresponding follow-up steps (postCommit, suspend, closeClean) should be skipped, and the exception is thrown out of the task-manager to thread
2) if commit failed with fenced errors, follow-up steps are also skipped (tasks state should be un-changed) and the task-migration exception is thrown out of the task-manager.